SegmentationModule.py

```diff
@@ -0,0 +1,238 @@
+    """This class represents pipeline module responsible for segmentation"""
+
+    # public:
+    def __init__(self):
+        """Initializes module"""
+        self.model = None
+
+    def exec_fit(self, x_train, x_valid, y_train, y_valid, checkpoint_file, epochs=30, batch_size=32):
+        """Creates model and launches fit. Trained model stored in self.model
+
+        Args:
+            x_train: train data tensor (number of samples, duration, embedding)
+            x_valid: validation data tensor
+            y_train: train data labels tensor
+            y_valid: validation data labels tensor
+            checkpoint_file: name of file where checkpoints will be saved
+            epochs: the number of epochs to fit
+            batch_size: the number of samples in a single batch to fit"""
+        assert len(x_train.shape) == 3, "X_train shape must be (samples, time, embeddings)"
+        assert len(x_valid.shape) == 3, "X_valid shape must be (samples, time, embeddings)"
+        assert len(y_train.shape) == 3, "Y_train shape must be (samples, time, 1)"
+        assert len(y_valid.shape) == 3, "Y_valid shape must be (samples, time, 1)"
+        assert isinstance(checkpoint_file, str) and checkpoint_file.endswith(".h5"), ("Checkpoint file must be string"
+                                                                                      "ends with .h5")
+        if self.model is None:  # build default model
+            self.__build_new_model()
+        y_tr_crf, y_val_crf = Models.data_to_crf(y_train, y_valid)  # convert to CRF-suitable format
+
+        callback_list = self.__define_callback_list(checkpoint_file)
+
+        self.model.fit(x_train, y_tr_crf, batch_size=batch_size, validation_data=(x_valid, y_val_crf), epochs=epochs,
+                       callbacks=callback_list)
+
+    def load_from_checkpoint(self, checkpoint_file):
+        """Loads trained model into self.model from checkpoint file
+
+        Args:
+            checkpoint_file: file *.h5 with model weights"""
+        assert isinstance(checkpoint_file, str) and checkpoint_file.endswith('.h5'), "Wrong checkpoint file argument"
+        self.model = Models.load_from_cpt(checkpoint_file)
+
+    def predict(self, x_data):
+        """Makes prediction on x_data. Model should be loaded or trained before predict called
+
+        Args:
+             x_data: tensor of embeddings to make prediction
+
+        Returns:
+            the list of time intervals where time interval is a list of the beginning time and the end time,
+            time stored in strings in format: 'hh:mm:ss'
+        """
+        assert self.model is not None, "Can't predict, model has not been loaded yet"
+        assert len(x_data.shape) == 3, "X_data shape must be (samples, duration, embeddings)"
+        # self.model must not be None
+        sample_mask = self.__get_raw_prediction(x_data)  # firstly, get raw prediction
+        smooth_mask = self.__get_smooth_mask(sample_mask)  # smooth mask
```

Comment:
add a flag for smoothing. or maybe create a possibility to change smoothing techniques